speech/cloud-client/transcribe_dlp.py

```diff
@@ -0,0 +1,140 @@
+    python transcribe_dlp.py deidentify -p <project_id>
+    python transcribe_dlp.py deidentify
+                            -f './resources/sallybrown.flac' -p <project_id>
+"""
+
+import argparse
+import io
+import re
+
+# Imports the Google Cloud client library
+from google.cloud import speech
+from google.cloud import dlp
+from google.cloud.speech import enums
+from google.cloud.speech import types
+
+
+def deidentify(file_name, project_id):
+    """ convert speech to text and then diedentify entities using dlp """
+    # Instantiates a client
+    speech_client = speech.SpeechClient()
+
+    dlp_client = dlp.DlpServiceClient()
+
+    parent = dlp_client.project_path(project_id)
+
+    # Prepare info_types by converting the list of strings into a list of
+    info_types = ['ALL_BASIC']
+    # dictionaries (protos are also accepted).
+    inspect_config = {
+        'info_types': [{'name': info_type} for info_type in info_types]
+    }
+    # Construct deidentify configuration dictionary
+    deidentify_config = {
+        'info_type_transformations': {
+            'transformations': [
+                {
+                    'primitive_transformation': {
+                        'replace_with_info_type_config': {
+
+                        }
+                    }
+                }
+            ]
+        }
+    }
+
+    # Loads the audio into memory
+    with io.open(file_name, 'rb') as audio_file:
+        content = audio_file.read()
+        audio = types.RecognitionAudio(content=content)
+
+    config = types.RecognitionConfig(
+        encoding=enums.RecognitionConfig.AudioEncoding.FLAC,
+        sample_rate_hertz=16000,
+        language_code='en-US')
+
+    # Detects speech in the audio file
+    response = speech_client.recognize(config, audio)
+
+    transcript = ""
+
+    for result in response.results:
+        transcript = transcript + result.alternatives[0].transcript
+
+    print('Original Transcript: {}'.format(transcript))
+
+    # Check transcription for email address,
+    # since speech API returns " at " instead of "@"
+    # Format with regex before sending to DLP api
+    # Currently social security numbers and credit card numbers
+    # are interpreted as phone numbers
+
+    regex = (
+             r".([A-Za-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[A-Za-z0-9!#$%&'*+/=?^_`"
+             r"{|}~-]+)*)(\sat\s+)((?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+"
+             r"[a-z0-9](?:[a-z0-9-]*[a-z0-9]))"
+    )
+
+    updated_transcript = re.sub(regex, r" \1@\3", transcript)
+
+    print('Email addresses reformatted: {}'.format(updated_transcript))
+
+    # Construct item
+    item = {'value': updated_transcript}
+
+    # Call the API
+    dlp_response = dlp_client.deidentify_content(
+        parent, inspect_config=inspect_config,
+        deidentify_config=deidentify_config, item=item)
+
+    # Print out the results.
+    print('Final Result with sensitive content redacted: {}'
+          .format(dlp_response.item.value))
```

Comment:
Nit (optional). I prefer to avoid hanging indent, this could also be written as:

```python
print('Final result with sensitive data redacted: {}'
    .format(dlp.response.item.value))
```